### ISSUE TITLE
Linked to Grid and Autostop behavior

### ISSUE THREAD

**Issue by skippskipp** [Tangerine-Community/Tangerine-client#174]
When a survey subtest is linked to a grid, the survey subtest should skip entirely if the grid's autostop setting has been triggered.  This only applies to grids which have a value set for 'Autostop.'  

In the LTTP2 assessment, the Task 5 grid has an autostop value of 6, meaning that if the first 6 grid items are marked incorrect, the grid should stop time and place the 'last attempted' bracket on item #6.  Whenever an autostop has been triggered in a case like this, the subsequent survey (linked) subtest should skip automatically.  Currently, following the autostop, the survey (linked) subtest appears.  It appears without any questions in it, but it should not appear at all.

**Comment by rjcorwin** [comment]
@skippskipp I believe I understand the intended behavior as seen on the databases site and I am reproducing what you describe using the current codebase.

**Comment by rjcorwin** [comment]
I checked out the code back to Dec 3 and saw that this regression existed then as well. In the above commit I rolled my own solution for this but it would be prudent to dig up code that had this working and study how it used to be done. The reason being I'm afraid that my code may be glossing over some things that also need to be done like marking those skipped subtests with some indicator in the results.
